### PR TITLE
DAOS-4185 control: Handle large pbin payloads

### PR DESCRIPTION
**Pull request by mjmac**
Adds pbin.ReadMessage() to handle payloads up to 1MiB in
size.

### CODE REVIEW

File: src/control/pbin/exec.go [hunk 1]
```diff
@@ -150,25 +188,29 @@ func ExecReq(parent context.Context, log logging.Logger, binPath string, req *Re
...
 		}
 
-		res, err = decodeResponse(recvData[:recvLen])
+		res, err = decodeResponse(recvData)
 		if err != nil && err != io.EOF {
-			log.Debugf("discarding garbage response %q", recvData[:recvLen])
-			readCount++
+			log.Debugf("discarding garbage response %q", recvData)
 			if readCount < maxReadAttempts {
+				readCount++
```
Comment (jbrosenz) [inline src/control/pbin/exec.go]: If you move readCount++ here, then it allows maxReadAttempts + 1 reads.  In this case, you'd allow 6, when maxReadAttemps said 5.

Comment (mjmac) [reply src/control/pbin/exec.go]: D'oh, you're right. When I was refactoring I moved where the increment happens and didn't notice this.

File: src/control/pbin/exec_test.go [hunk 2]
```diff
@@ -38,35 +40,78 @@ import (
...
+	post := `"}`
+	adjSize := size - len(post)
+
+	for buf.Len() < adjSize {
+		buf.WriteString("0123456789abcdefghijklmnopqrstuvxyz")
+	}
+
+	ret := buf.Bytes()
+	if len(ret) > adjSize {
+		ret = ret[:len(ret)-(len(ret)-(adjSize))]
```
Comment (kjacque) [inline src/control/pbin/exec_test.go]: Doesn't that simplify down to adjSize?

Comment (mjmac) [reply src/control/pbin/exec_test.go]: ... yes. Not really sure what I was thinking there. Fixed, thanks.

**Comment by mjmac** [comment]
Ugh, sorry. One last time. Really distracted today.

**Comment by mjmac** [comment]
Functional failure was [DAOS-4153](https://jira.hpdd.intel.com/browse/DAOS-4153).

**Pull request merged**